tests/app/questionnaire/test_navigation.py

```diff
@@ -518,3 +518,128 @@ def test_navigation_repeating_group_link_name_format(self):
         ]
 
         self.assertEqual(navigation.build_navigation('property-details', 0), user_navigation)
+
+    def test_navigation_skip_condition_hide_group(self):
+        survey = load_schema_file("test_navigation.json")
+
+        metadata = {
+            "eq_id": '1',
+            "collection_exercise_sid": '999',
+            "form_type": "some_form"
+        }
+
+        completed_blocks = []
+
+        answer_store = AnswerStore()
+
+        answer_1 = Answer(
+            value="Contents",
+            group_instance=0,
+            block_id='insurance-type',
+            group_id='property-details',
+            answer_instance=0,
+            answer_id='insurance-type-answer'
+        )
+
+        answer_store.add(answer_1)
+
+        navigation = Navigation(survey, answer_store, metadata, completed_blocks=completed_blocks)
+
+        user_navigation = [
+            {
+                'repeating': False,
+                'highlight': True,
+                'link_name': 'Property Details',
+                'completed': False,
+                'link_url': 'http://test/questionnaire/1/some_form/999/property-details/0/insurance-type'
+            },
+            {
+                'repeating': False,
+                'highlight': False,
+                'link_name': 'Household Details',
+                'completed': False,
+                'link_url': 'http://test/questionnaire/1/some_form/999/multiple-questions-group/0/household-composition'
+            },
+            {
+                'repeating': False,
+                'highlight': False,
+                'link_name': 'Extra Cover',
+                'completed': False,
+                'link_url': 'http://test/questionnaire/1/some_form/999/extra-cover/0/extra-cover-block'
+            },
+            {
+                'repeating': False,
+                'highlight': False,
+                'link_name': 'Payment Details',
+                'completed': False,
+                'link_url': 'http://test/questionnaire/1/some_form/999/payment-details/0/credit-card'
+            }
+        ]
+
+        self.assertEqual(navigation.build_navigation('property-details', 0), user_navigation)
```

Comment:
I think it would be clearer to test that the `House Details` navigation item doesn't exist, rather than testing for a match of the whole navigation structure. Something like:

```
navigation = Navigation(survey, answer_store, metadata, completed_blocks=completed_blocks)
user_navigation = navigation.build_navigation('property-details', 0)
link_names = [d['link_name'] for d in user_navigation]
self.assertFalse('House Details' in link_names)
```